mmdet/models/losses/focal_loss.py

```diff
@@ -37,7 +37,14 @@ def sigmoid_focal_loss(pred,
     loss = _sigmoid_focal_loss(pred, target, gamma, alpha)
     # TODO: find a proper way to handle the shape of weight
     if weight is not None:
-        weight = weight.view(-1, 1)
+        if weight.shape != loss.shape:
+            if weight.size(0) == loss.size(0):
+                # weight does not have num_class dim
+                weight = weight.view(-1, 1)
+            else:
+                # weight is flattened while loss is not
```

Comment:
We may also note what kind of method would encounter such a situation.

Reply:
This is because the weight is flattened in the default loss_single function in AnchorHead. 
see https://github.com/open-mmlab/mmdetection/blob/c40cafb1e1347645c1289634a40cc26b7b58163c/mmdet/models/anchor_heads/anchor_head.py#L145

Reply:
Yes, I mean we could add they in the comment, e.g. `in Retinanet blablabla`, `in FSAF blablabla`

Reply:
Get it!